Sources/System/FileControlRaw.swift

```diff
@@ -0,0 +1,726 @@
+    @_alwaysEmitIntoClient
+    public static var addFileSignaturesForDYLDSim: Command {
+      Command(F_ADDFILESIGS_FOR_DYLD_SIM)
+    }
+
+    /// Fsync + issue barrier to drive.
+    ///
+    /// The corresponding C constant is `F_BARRIERFSYNC`.
+    @_alwaysEmitIntoClient
+    public static var barrierFsync: Command {
+      Command(F_BARRIERFSYNC)
+    }
+
+    /// Add signature from same file, return end offset in structure on success.
+    ///
+    /// The corresponding C constant is `F_ADDFILESIGS_RETURN`.
+    @_alwaysEmitIntoClient
+    public static var addFileSignaturesReturn: Command {
+      Command(F_ADDFILESIGS_RETURN)
+    }
+
+    /// Check if Library Validation allows this Mach-O file to be mapped into
+    /// the calling process.
+    ///
+    /// The corresponding C constant is `F_CHECK_LV`.
+    @_alwaysEmitIntoClient
+    public static var checkLibraryValidation: Command {
+      Command(F_CHECK_LV)
+    }
+
+    /// Deallocate a range of the file.
+    ///
+    /// The corresponding C constant is `F_PUNCHHOLE`.
+    @_alwaysEmitIntoClient
+    public static var punchhole: Command { Command(F_PUNCHHOLE) }
+
+    /// Trim an active file.
+    ///
+    /// The corresponding C constant is `F_TRIM_ACTIVE_FILE`.
+    @_alwaysEmitIntoClient
+    public static var trimActiveFile: Command {
+      Command(F_TRIM_ACTIVE_FILE)
+    }
+
+    /// Synchronous advisory read fcntl for regular and compressed file.
+    ///
+    /// The corresponding C constant is `F_SPECULATIVE_READ`.
+    @_alwaysEmitIntoClient
+    public static var speculativeRead: Command {
+      Command(F_SPECULATIVE_READ)
+    }
+
+    /// Return the full path without firmlinks of the fd.
+    ///
+    /// Note: A Swiftier wrapper is `FileDescriptor.getPath(_:)`.
+    ///
+    /// The corresponding C constant is `F_GETPATH_NOFIRMLINK`.
+    @_alwaysEmitIntoClient
+    public static var getPathNoFirmLink: Command {
+      Command(F_GETPATH_NOFIRMLINK)
+    }
+
+    /// Add signature from same file, return information.
+    ///
+    /// The corresponding C constant is `F_ADDFILESIGS_INFO`.
+    @_alwaysEmitIntoClient
+    public static var addFileSignatureInfo: Command {
+      Command(F_ADDFILESIGS_INFO)
+    }
+
+    /// Add supplemental signature from same file with fd reference to original.
+    ///
+    /// The corresponding C constant is `F_ADDFILESUPPL`.
+    @_alwaysEmitIntoClient
+    public static var addFileSupplementalSignature: Command {
+      Command(F_ADDFILESUPPL)
+    }
+
+    /// Look up code signature information attached to a file or slice.
+    ///
+    /// The corresponding C constant is `F_GETSIGSINFO`.
+    @_alwaysEmitIntoClient
+    public static var getSignatureInfo: Command {
+      Command(F_GETSIGSINFO)
+    }
+
+    /// Shared or read lock.
+    ///
+    /// The corresponding C constant is `F_RDLCK`.
+    @_alwaysEmitIntoClient
+    public static var readLock: Command { Command(F_RDLCK) }
```

Comment:
Wait, are `F_RDLCK`/`F_WRLCK`/`F_UNLCK` really fcntl commands?

Reply:
They're in the "Command" namespace but they're used as parameters. These grab-bag APIs don't have clear orthogonal namespaces. We could consider separating them out. There's a tradeoff between spending our design/weirdness budget on making these APIs better and directly surfacing the interfaces as-is. For now, I've been promoting the most important/common to `FileDescirptor` namespace, while the obscure or specialized options try to mirror `fcntl` as closely as possible.